latch/ldata/path.py

```diff
@@ -0,0 +1,271 @@
+
+    def size(self, *, load_if_missing: bool = True) -> Optional[int]:
+        if self._cache.size is None and load_if_missing:
+            self.load_metadata()
+        return self._cache.size
+
+    def content_type(self, *, load_if_missing: bool = True) -> Optional[str]:
+        if self._cache.content_type is None and load_if_missing:
+            self.load_metadata()
+        return self._cache.content_type
+
+    def is_dir(self, *, load_if_missing: bool = True) -> bool:
+        return self.type(load_if_missing=load_if_missing) in dir_types
+
+    def iterdir(self) -> Generator["LPath", None, None]:
+        """Yield LPaths objects contained within the directory.
+
+        Should only be called on directories. Does not recursively list directories.
+
+        Always makes a network request.
+        """
+        data = execute(
+            gql.gql("""
+            query LDataChildren($argPath: String!) {
+                ldataResolvePathData(argPath: $argPath) {
+                    finalLinkTarget {
+                        type
+                        childLdataTreeEdges(filter: { child: { removed: { equalTo: false } } }) {
+                            nodes {
+                                child {
+                                    name
+                                }
+                            }
+                        }
+                    }
+                }
+            }"""),
+            {"argPath": self.path},
+        )["ldataResolvePathData"]
+
+        if data is None:
+            raise FileNotFoundError(f"No such Latch file or directory: {self.path}")
+        if data["finalLinkTarget"]["type"].lower() not in dir_types:
+            raise ValueError(f"{self.path} is not a directory")
+
+        for node in data["finalLinkTarget"]["childLdataTreeEdges"]["nodes"]:
+            yield LPath(urljoins(self.path, node["child"]["name"]))
+
+    def rmr(self) -> None:
+        """Recursively delete files at this instance's path.
+
+        Always makes a network request.
+        """
+        execute(
+            gql.gql("""
+            mutation LDataRmr($nodeId: BigInt!) {
+                ldataRmr(input: { argNodeId: $nodeId }) {
+                    clientMutationId
+                }
+            }
+            """),
+            {"nodeId": self.node_id},
+        )
+
+    def copy_to(self, dst: "LPath", *, show_summary: bool = False) -> None:
+        _remote_copy(self.path, dst.path, show_summary=show_summary)
+
+    def upload_from(self, src: Path, *, show_progress_bar: bool = False) -> None:
+        _upload(
+            os.fspath(src),
+            self.path,
+            progress=Progress.tasks if show_progress_bar else Progress.none,
+            verbose=False,
+        )
+
+    def download(
+        self, dst: Optional[Path] = None, *, show_progress_bar: bool = False
+    ) -> Path:
+        if dst is None:
+            global download_idx
+            dir = Path.home() / "lpath"
```

Comment:
just realized that we want to support normal people's computers and not just workflows

```suggestion
            dir = Path.home() / ".latch" / "lpath"
```

1. we probably actually wanna use some XDG stuff here idk. this is a bad place to put stuff on user-facing Linux
2. we also probably want to use temporary files somehow